floss/language/go/extract.py

```diff
@@ -0,0 +1,408 @@
+        addr = m.start() + pe.OPTIONAL_HEADER.ImageBase + section_va
+        try:
+            string = m.group(0).decode("utf-8")
+            yield StaticString(string=string, offset=addr, encoding=StringEncoding.ASCII)
+        except UnicodeDecodeError:
+            continue
+        break
+
+
+def extract_string_blob(pe: pefile.PE, section_data, section_va, min_length) -> Iterable[StaticString]:
+    # Extract string blob in .rdata section
+    """
+    0048E620  5B 34 5D 75 69 6E 74 38  00 09 2A 5B 38 5D 69 6E  [4]uint8..*[8]in
+    0048E630  74 33 32 00 09 2A 5B 38  5D 75 69 6E 74 38 00 09  t32..*[8]uint8..
+    0048E640  2A 5B 5D 73 74 72 69 6E  67 00 09 2A 5B 5D 75 69  *[]string..*[]ui
+    0048E650  6E 74 31 36 00 09 2A 5B  5D 75 69 6E 74 33 32 00  nt16..*[]uint32.
+    0048E660  09 2A 5B 5D 75 69 6E 74  36 34 00 09 2A 63 68 61  .*[]uint64..*cha
+    0048E670  6E 20 69 6E 74 01 09 41  6E 6F 6E 79 6D 6F 75 73  n int..Anonymous
+    0048E680  01 09 43 61 6C 6C 53 6C  69 63 65 01 09 43 6C 65  ..CallSlice..Cle
+    0048E690  61 72 42 75 66 73 01 09  43 6F 6E 6E 65 63 74 45  arBufs..ConnectE
+    0048E6A0  78 01 09 46 74 72 75 6E  63 61 74 65 01 09 49 6E  x..Ftruncate..In
+    0048E6B0  74 65 72 66 61 63 65 01  09 4E 75 6D 4D 65 74 68  terface..NumMeth
+    0048E6C0  6F 64 01 09 50 72 65 63  69 73 69 6F 6E 01 09 52  od..Precision..R
+    """
+
+    blob_pattern = re.compile(b"(\x00|\x01)(?P<blob>.)", re.DOTALL)
+    for m in blob_pattern.finditer(section_data):
+        if m.group("blob") != b"\x00":
+            data = section_data[m.end() : m.end() + m.group(2)[0]]
+            try:
+                data = data.decode("utf-8")
+                if str(data).isprintable() and data != "" and len(data) >= min_length:
+                    addr = m.start() + pe.OPTIONAL_HEADER.ImageBase + section_va
+                    yield StaticString(string=data, offset=addr, encoding=StringEncoding.ASCII)
+
+            except UnicodeDecodeError:
+                continue
+
+
+def extract_string_blob2(pe: pefile.PE, section_data, section_va, min_length) -> Iterable[StaticString]:
+    # Extract string blob in .rdata section that starts with "go:buildid" or "go.buildid"
+    """
+    67 6F 3A 62 75 69 6C 64  69 64 00 69 6E 74 65 72  go:buildid.inter
+    6E 61 6C 2F 63 70 75 2E  49 6E 69 74 69 61 6C 69  nal/cpu.Initiali
+    7A 65 00 69 6E 74 65 72  6E 61 6C 2F 63 70 75 2E  ze.internal/cpu.
+    70 72 6F 63 65 73 73 4F  70 74 69 6F 6E 73 00 69  processOptions.i
+    6E 74 65 72 6E 61 6C 2F  63 70 75 2E 69 6E 64 65  nternal/cpu.inde
+    78 42 79 74 65 00 69 6E  74 65 72 6E 61 6C 2F 63  xByte.internal/c
+    """
+
+    blob_pattern = re.compile(b"go(\x2E|\x3A)buildid\x00(.)*\x00\x00", re.DOTALL)
+    for m in blob_pattern.finditer(section_data):
+        t = m.group(0)
+        for s in t.split(b"\x00"):
+            try:
+                x = s.decode("utf-8")
+                if x.isprintable() and x != "" and len(x) >= min_length:
```

Comment:
here's still this extra check (!= "") we removed at other places, would it make sense to turn this part into a helper function?